### PR TITLE
Logger

### LINKED ISSUE #16: Implement Logger Library
Implement a logger library.

### PR DESCRIPTION
**Pull request by AndreasLrx**
Add a logger with multiple severities (Debug, info, warning, error).
It can add the time, date, PID and thread ID when writing a message.

In the project, the logger is set with a loglevel of info, displays only the time of the message and is linked to the file "log.txt" at the root (see main.cpp).

Close #16

### CODE REVIEW

**Comment by AndreasLrx** [comment]
Need #21 validation before.

**Review by ZQUEMA — APPROVED** [review]
LGTM

File: src/logger/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,13 @@
+set(INCROOT ${INCROOT}/logger)
+set(SRCROOT ${SRCROOT}/logger)
+
+set(SRC
+	${SRC}
+    ${SRCROOT}/Logger.cpp
```
Comment (MisterPeModder) [inline src/logger/CMakeLists.txt:6]: You used the `tab` character for indentation in this file, but used spaces for this line.
Is this normal?

File: src/logger/Logger.hpp [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+    /// @return reference to the current value of the parameter.
+    bool &operator[](LogInfo logInfo);
+
+    /// Set the display style of the logger.
+    ///
+    /// @param[in] time If the time must be displayed "hh:dd:ss"
+    /// @param[in] date If the date must be displayed "dd/mm/yy"
+    /// @param[in] pid  If the process PID must be displayed "PID(pid)"
+    /// @param[in] threadId If the thread id must be displayed "ThreadID(id)"
+    void setStyle(bool time = true, bool date = false, bool pid = false, bool threadId = false);
```
Comment (MisterPeModder) [inline src/logger/Logger.hpp:126]: Is this method necessary? I feel like calls to this method may be hard to understand; a call like:
```cpp
logger.setStyle(true, false, true, false);
```
Is much harder to read than:
```cpp
logger.setStyle(Logger::LogInfo::PID, Logger::LogInfo::Time);
```

Moreover, this style of function doesn't lend very well to extensions, what happens when you want to add a new style?

File: src/main.cpp [hunk 3]
```diff
@@ -1,3 +1,14 @@
...
+{
+    /// Setup the logger parameters
+    Logger::logger.setOutputFile("log.txt");
+    Logger::logger.setLogLevel(Logger::Severity::Information);
+    Logger::logger.setStyle(true, false, false, false);
+
+    Logger::logger.log(Logger::Severity::Information, "Start of program");
+    std::cout << "Hello, World!" << std::endl;
+    Logger::logger.log(Logger::Severity::Information, "End of program");
+}
```
Comment (MisterPeModder) [inline src/main.cpp:14]: Missing an empty line here.

**Review by MisterPeModder — CHANGES_REQUESTED** [review]
The logging looks very solid overall, I just have questions about the Logger::setStyle() method.

File: src/logger/CMakeLists.txt (hunk 1 above)
Comment (AndreasLrx) [reply src/logger/CMakeLists.txt]: It wasn't normal, fixed in last commit.

**Review by MisterPeModder — CHANGES_REQUESTED** [review]

**Review by aurelenc — APPROVED** [review]

**Pull request merged**